include/odp/api/spec/schedule.h

```diff
@@ -368,6 +368,27 @@ void odp_schedule_order_lock(unsigned lock_index);
 void odp_schedule_order_unlock(unsigned lock_index);
 
 /**
+ * Release existing ordered context lock and acquire a new lock
+ *
+ * This call is valid only when holding an ordered synchronization context.
+ * Release a previously locked ordered context lock and acquire a new ordered
+ * context lock. This call is valid only when there is a single ordered context
+ * lock active. Results are undefined if this call is made when multiple ordered
+ * context locks are acquired in nested fashion.
+ * This call does not release the existing ordered context and only switches
+ * from one ordered lock to another ordered lock within the same ordered
+ * context.
+ *
+ * @param lock_index	Index of the ordered lock in the current context to be
+ *			acquired. Previously acquired ordered lock is released.
+ *			Must be in the range 0...odp_queue_lock_count() - 1.
```

Comment:
Shouldn't lock_index be in the range 1..odp_queue_lock_count()-1? We are supposedly already holding one lock (lock_index >=0), this lock will be released and the next lock will be acquired.

Reply:
ODP does not have any architectural requirement that ordered locks be acquired in any particular sequence, or that they be acquired at all. The only requirement is that they only can be acquired once per ordered context. So this wording is correct. It's entirely possible that a thread acquires lock 1 first and then switches to lock 0. It's entirely up to the application. 

The beauty of ordered locks is that they are deadlock-free by design even in the face of random acquisition order among the various threads that use them.

Reply:
Just to reiterate, ordered locks are used to make sure packet belonging to a particular flow acquire the lock in order. There is no need to acquire the lock in ascending order. Even though our linux-generic implementation is using them as index number for performance reasons it will usually be the last few bits of the flow id calculated by RSS or any other mechanism decided by the application.